### ISSUE TITLE
Add overflow:hidden to the .side class to get rid of horizontal scroll bars

### ISSUE THREAD

**Issue by anjin** [reddit/reddit#1491]
This is a two second fix that I can't believe hasn't been implemented. Please add overflow hidden to the .side class so that too-wide content like long flairs doesn't break the page layout.

Without overflow:hidden you get horizontal scrollbars:
<img width="1498" alt="screen shot 2015-12-30 at 2 05 42 pm" src="https://cloud.githubusercontent.com/assets/2884/12058232/d43d3bf0-aefe-11e5-8917-31f9f6b5d017.png">

With overflow:hidden everything is contained:
<img width="1498" alt="screen shot 2015-12-30 at 2 05 57 pm" src="https://cloud.githubusercontent.com/assets/2884/12058244/f188e042-aefe-11e5-96b1-c0a395b63c64.png">

**Comment by 13steinj** [comment]
This is actually OS and browser dependant, sometimes browsers render the page without the excess anyway.

But this isn't really a solution. `overflow-x: hidden` would hide some things, and `overflow-x: scroll`, IIRC, wasn't liked before (scrollbar on just the side div).

**Comment by anjin** [comment]
The content inside the .side class div is breaking the page layout when the
content is wider than the set width of the sidebar. There are many things
inside the sidebar that can be done to make sure that content doesn't break
outside the bounding box of the .side, but the easiest thing to do would be
to hide anything that overflows so it doesn't affect the entire page
layout.

It's pretty janky to just not apply a simple global solution to prevent
breaking the layout.

On Thursday, December 31, 2015, 13steinj <notifications@github.com> wrote:

> This is actually OS and browser dependant, sometimes browsers render the
> page without the excess anyway.
>
> But this isn't really a solution. overflow-x: hidden would hide some
> things, and overflow-x: scroll, IIRC, wasn't liked before (scrollbar on
> just the side div).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/reddit/reddit/issues/1491#issuecomment-168263348>.
>


-- 
Lee Donahoe
*LEMUR Heavy Industries*
*p 323.275.4588 | c 310.721.1510 | lemurheavy.com <http://lemurheavy.com>*

**Comment by jkingsman** [comment]
An alternative solution: `word-break: break-word;` on `.side` allows mid-word breaking on 'unbreakable' words (i.e. longer than the box without a space or other breakable spot). No unsightly scrollbars, nothing gets clipped, and the page doesn't get broken. If CSS can sanely break things, it will; otherwise, it will break at the nearest character that will still respect the width.

Adding something that affects word breaking on all sidebars is a pretty sweeping change, but a broken layout on a not-difficult-to-encounter situation is pretty sweeping as well.

**Comment by 13steinj** [comment]
@jkingsman Would that work in cases that an extremely wide table is within the sidebar? I don't want to assume, but I don't think so. (Ex r/IAMA without custom subreddit styles enabled)

**Comment by jkingsman** [comment]
Yes, it will break the cell contents and respect width. I copied the table from r/IAmA into r/test's sidebar (with no custom CSS) and things looked 'fine' (I quote that because it's not always pretty how it decides to break).

### Before `word-break: break-word;`: 
![before](https://cloud.githubusercontent.com/assets/2365891/12220266/39a8e812-b720-11e5-8f11-bfd18c20fe89.PNG)

### After `word-break: break-word;`:
![after](https://cloud.githubusercontent.com/assets/2365891/12220265/39a61560-b720-11e5-9e13-d24bbbb667e5.PNG)
